android-components/components/browser/engine-gecko/src/test/java/mozilla/components/test/ReflectionUtils.kt

```diff
@@ -4,18 +4,38 @@
 
 package mozilla.components.test
 
+import java.lang.invoke.MethodHandles
 import java.lang.reflect.Field
 import java.lang.reflect.Modifier
+import java.security.AccessController
+import java.security.PrivilegedAction
 
 object ReflectionUtils {
     fun <T : Any> setField(instance: T, fieldName: String, value: Any?) {
-        val originField = instance.javaClass.getField(fieldName)
+        val mapField = AccessController.doPrivileged(
+            PrivilegedAction {
+                try {
+                    val field = instance::class.java.getField(fieldName)
+                    field.isAccessible = true
+                    field.removeFinal()
+                    return@PrivilegedAction field
+                } catch (e: ReflectiveOperationException) {
+                    throw Error(e)
+                }
+            },
+        )
 
-        val modifiersField = Field::class.java.getDeclaredField("modifiers")
-        modifiersField.isAccessible = true
-        modifiersField.setInt(originField, originField.modifiers and Modifier.FINAL.inv())
+        mapField.set(instance, value)
+    }
+}
 
-        originField.isAccessible = true
-        originField.set(instance, value)
+private fun Field.removeFinal() {
```

Comment:
Stub comment: I don't understand this enough to review it yet but I don't want to hold up the rest of the review comments while I do some research.

Reply:
seems this not even needed.